golang/cosmos/go.mod

```diff
@@ -192,9 +192,6 @@ replace (
 	// Pick up an IAVL race fix.
 	github.com/cosmos/iavl => github.com/cosmos/iavl v0.19.7
 
-	// Async version negotiation
-	github.com/cosmos/ibc-go/v6 => github.com/agoric-labs/ibc-go/v6 v6.2.1-alpha.agoric.3
```

Comment:
@michaelfig should confirm, but looking at the [diff of v6.2.1-alpha.agoric.3 from upstream v6.2.1](https://github.com/cosmos/ibc-go/compare/v6.2.1...agoric-labs:ibc-go:v6.2.1-alpha.agoric.3#diff-b225d8b78960f1ed19ccabcbdcda5ff708075790adac8463fbb0a83a8a63b78a), I don't think we can move away from our fork. And there are hopefully tests in agoric-sdk that will fail if we try.

Reply:
> I don't think we can move away from our fork. And there are hopefully tests in agoric-sdk that will fail if we try.

@LuqiPan's changes in this PR remove the async version negotiation feature detection between our ibc-go fork and agoric-sdk, which allows the tests to pass again while preventing agoric-sdk from using the AsyncVersion code.

In a meeting some time ago, we had agreed not to use that semi-automatic feature detection, and instead would document in the interchain stack upgrade runbook the conditions under which we could set `AsyncVersions = true`, namely by performing some investigation into the ibc-go implementation we rely on before landing PRs like this one.